pkg/csi/service/wcpguest/controller_helper.go

```diff
@@ -275,15 +276,49 @@ func isPVCInSupervisorClusterBound(ctx context.Context, client clientset.Interfa
 		if !ok {
 			continue
 		}
-		log.Debugf("PersistentVolumeClaim %s in namespace %s is in state %s. Received event %v",
-			pvcName, ns, pvc.Status.Phase, event)
-		if pvc.Status.Phase == v1.ClaimBound && pvc.Name == pvcName {
-			log.Infof("PersistentVolumeClaim %s in namespace %s is in state %s", pvcName, ns, pvc.Status.Phase)
-			return true, nil
+
+		if pvc.Name == supervisorPvcName {
+			if pvc.Status.Phase == v1.ClaimBound {
+				log.Infof("PersistentVolumeClaim %s in namespace %s is in state %s", supervisorPvcName, supervisorPvcNamespace, pvc.Status.Phase)
+				return true, nil
+			} else if pvc.Status.Phase == v1.ClaimPending {
+				var failureMessage string
+				_ = wait.PollImmediate(2*time.Second, 1*time.Minute, func() (done bool, err error) {
```

Comment:
Can we move `wait.PollImmediate` outside and make a call to get PV status, if PV status is bound return else get list of events and see if failure is present, if yes then return failure.

we can remove `client.CoreV1().PersistentVolumeClaims(ns).Watch(timeoutSeconds, supervisorPvcName, supervisorPvcNamespace, v1.ClaimBound)`  and replace it with one loop for `wait.PollImmediate`, as failure to provision volume will not get notified using watch.

Also with your implementation, we will always go into the pending check first, and wait for 1 minute to capture events, so for positive case, even when the supervisor is provisioning volume within seconds, we are delaying it in the guest.